src/metpy/testing.py

```diff
@@ -23,37 +25,89 @@
 from .deprecation import MetpyDeprecationWarning
 from .units import units
 
-MPL_VERSION = Version(matplotlib.__version__)
 
+def version_check(version_spec):
+    """Return comparison between the active module and a requested version number.
 
-def mpl_version_before(ver):
-    """Return whether the active matplotlib is before a certain version.
+    Will also validate specification against package metadata to alert if spec is irrelevant.
 
     Parameters
     ----------
-    ver : str
-        The version string for a certain release
+    version_spec : str
+        Module version specification to validate against installed package. Must take the form
+        of `f'{module_name}{comparison_operator}{version_number}'` where `comparison_operator`
+        must be one of `['==', '=', '!=', '<', '<=', '>', '>=']`, eg `'metpy>1.0'`.
 
     Returns
     -------
-        bool : whether the current version was released before the passed in one
+        bool : Whether the installed package validates against the provided specification
     """
-    return MPL_VERSION < Version(ver)
+    comparison_operators = {
+        '==': op.eq, '=': op.eq, '!=': op.ne,
+        '<': op.lt, '<=': op.le,
+        '>': op.gt, '>=': op.ge,
+    }
```

Comment:
Intentionally keeping this arranged like this? Otherwise could compress the lines here now that it's shorter.

Reply:
An exact fit!